eclipse-theia/theia
# New File opens untitled editor

#### What it does


Fixes #3354 by building on #10608, and moving the code necessary to handle untitled resources into `core` rather than `plugin-ext`.

#### How to test


1. The New file keyboard shortcut and menu item should open an untitled editor without prompting immediately for a filename.
2. You should be able to use that editor to type whatever you'd like.
3. If you try to close the widget after adding content, it should prompt you to save
4. If you try to save the widget, it should open a file picker with a location determined by the following considerations (at the time you created the widget)
     - If you had another editor open at the time, it should try to place the new file in the same folder.
     - If you had a URI selected in the navigator, it should try to place the file in that location (same folder if file, in the folder if a folder is selected)
     - If you have a workspace open, it should try to place the file in the (first) root
     - It should fall back to your home directory



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
